src/core/dbus-execute.c

```diff
@@ -394,14 +394,33 @@ static int property_get_syscall_filter(
                 return r;
 
 #if HAVE_SECCOMP
-        SET_FOREACH(id, c->syscall_filter, i) {
-                char *name;
+        HASHMAP_FOREACH_KEY(val, id, c->syscall_filter, i) {
+                _cleanup_free_ char *name = NULL;
+                const char *e = NULL;
+                char *s;
+                int num = PTR_TO_INT(val);
 
                 name = seccomp_syscall_resolve_num_arch(SCMP_ARCH_NATIVE, PTR_TO_INT(id) - 1);
                 if (!name)
                         continue;
 
-                r = strv_consume(&l, name);
+                if ((num & 0xff) == num) {
```

Comment:
<strike>Hmm, why would `num` have a value above 0xff? I thought you check in the code that assigns those values that they are in that range. Maybe an assert would be in order instead?</strike>

Ah, OK, so you're checking if `num >= 0`. I'd find it much less confusing if written as `if (num >= 0)`.

Reply:
OK. Changed to `(num >= 0 && num <= 255)`.